context/sm_context.go

```diff
@@ -724,7 +724,9 @@ func (smContext *SMContext) PublishSmCtxtInfo() {
 
 	//Populate kafka sm ctxt struct
 	kafkaSmCtxt.Imsi = smContext.Supi
-	kafkaSmCtxt.IPAddress = smContext.PDUAddress.Ip.String()
+	if smContext.PDUAddress.Ip != nil {
+		kafkaSmCtxt.IPAddress = smContext.PDUAddress.Ip.String()
```

Comment:
in case the IP is nil what will the value of kafkaSmCtxt.IPAddress be ?

Reply:
Do we need to check PDUAddress also to be nil ?

Reply:
THanks. Resolved the issue.